lib/rules/selector-max-attribute/index.mjs

```diff
@@ -92,8 +90,21 @@ const rule = (primary, secondaryOptions) => {
 			if (!isStandardSyntaxRule(ruleNode)) return;
 
 			flattenNestedSelectorsForRule(ruleNode, result).forEach(({ selector, resolvedSelectors }) => {
-				resolvedSelectors.forEach((resolvedSelector) => {
-					checkSelector(resolvedSelector, selector, ruleNode);
+				resolvedSelectors.walk((childSelector) => {
```

Comment:
[note] I wish `postcss-selector-parser` would add `walkSelector` like the existing `walkPseudos`! 😄

Reply:
I was also surprised that it was missing!